latitudesh/lsh
# Chore: adjustments to flags

### What does this PR do?
This PR aims to improve the general usability of the CLI

#### Usability problems we aim to solve:
- Some endpoints require the user to set the id field twice, once in the URL and once in the request body. Ideally, the user should only need to send it once.
- Some endpoints require a type parameter that is always the same. We can set it automatically to improve the user experience.
- All endpoints display the API-Version flag. Currently, this flag serves no purpose as the API version is set globally. Also, setting it to send requests to v2 would break some of our models.

####  Example:
running `lsh servers update -h` right now will return:
```
Update server information.

Usage:
  lsh servers update [flags]

Flags:
      --API-Version string    (default "2023-06-01")
      --billing string       
      --body string          Optional json string for [body]. 
  -h, --help                 help for update
      --hostname string      
      --id string            
      --server_id string     Required. 
      --type string          Enum: ["servers"].
...

```
With this PR it will return
```
Update server information.

Usage:
  lsh servers update [flags]

Flags:
      --billing string    
      --body string       Optional json string for [body]. 
  -h, --help              help for update
      --hostname string   
      --id string         The Server Id (Required).
...
```
### How to Test:
You can verify all the commands help menu and it should have only one `--id` flag and no `--API-Version` and `--type` flags.
You can also test the endpoints to verify if they still work as intended.

### Endpoints improved
- [x] Servers and Reinstall
- [x] SSH Keys
- [x] API Key
- [x] Virtual Networks
- [x] Virtual Networks assignments
- [x] Plans
